Sources/SwiftGRPCNIOInteroperabilityTests/InteroperabilityTestCases.swift

```diff
@@ -0,0 +1,679 @@
+
+    try assertEqual(call1.response.map { $0.payload }, call2.response.map { $0.payload })
+    try assertEqual(call1.status.map { $0.code }, .ok)
+    try assertEqual(call2.status.map { $0.code }, .ok)
+  }
+}
+
+/// This test verifies unary calls succeed in sending messages, and touches on flow control (even
+/// if compression is enabled on the channel).
+///
+/// Server features:
+/// - UnaryCall
+///
+/// Procedure:
+/// 1. Client calls UnaryCall with:
+///    ```
+///    {
+///        response_size: 314159
+///        payload:{
+///            body: 271828 bytes of zeros
+///        }
+///    }
+///    ```
+///
+/// Client asserts:
+/// - call was successful
+/// - response payload body is 314159 bytes in size
+/// - clients are free to assert that the response payload body contents are zero and comparing
+///   the entire response message against a golden response
+class LargeUnary: InteroperabilityTest {
+  func run(using connection: GRPCClientConnection) throws {
+    let client = Grpc_Testing_TestServiceService_NIOClient(connection: connection)
+
+    let request = Grpc_Testing_SimpleRequest.with { request in
+      request.responseSize = 314_159
+      request.payload = .zeros(count: 271_828)
+    }
+
+    let call = client.unaryCall(request)
+
+    try assertEqual(call.response.map { $0.payload.body.count }, 314_159)
+    try assertEqual(call.status.map { $0.code }, .ok)
+  }
+}
+
+/// This test verifies that client-only streaming succeeds.
+///
+/// Server features:
+/// - StreamingInputCall
+///
+/// Procedure:
+/// 1. Client calls StreamingInputCall
+/// 2. Client sends:
+///    ```
+///    {
+///        payload:{
+///            body: 27182 bytes of zeros
+///        }
+///    }
+///    ```
+/// 3. Client then sends:
+///    ```
+///    {
+///        payload:{
+///            body: 8 bytes of zeros
+///        }
+///    }
+///    ```
+/// 4. Client then sends:
+///    ```
+///    {
+///        payload:{
+///            body: 1828 bytes of zeros
+///        }
+///    }
+///    ```
+/// 5. Client then sends:
+///    ```
+///    {
+///        payload:{
+///            body: 45904 bytes of zeros
+///        }
+///    }
+///    ```
+/// 6. Client half-closes
+///
+/// Client asserts:
+/// - call was successful
+/// - response aggregated_payload_size is 74922
+class ClientStreaming: InteroperabilityTest {
+  func run(using connection: GRPCClientConnection) throws {
+    let client = Grpc_Testing_TestServiceService_NIOClient(connection: connection)
+    let call = client.streamingInputCall()
+
+    call.newMessageQueue().flatMap {
+      call.sendMessage(.withPayload(of: .zeros(count: 27_182)))
+    }.flatMap {
+      call.sendMessage(.withPayload(of: .zeros(count: 8)))
+    }.flatMap {
+      call.sendMessage(.withPayload(of: .zeros(count: 1_828)))
+    }.flatMap {
+      call.sendMessage(.withPayload(of: .zeros(count: 45_904)))
+    }.whenSuccess {
+      call.sendEnd(promise: nil)
+    }
```

Comment:
Should we wait on this _before_ waiting on the response and status below?

Reply:
Good idea!